key.core/src/main/java/de/uka/ilkd/key/logic/Semisequent.java

```diff
@@ -29,15 +30,31 @@ private Semisequent() {
 
 
     /**
-     * creates a new Semisequent with the Semisequent elements in seqList; the provided list must be
-     * redundancy free, i.e., the created sequent must be exactly the same as when creating the
+     * Create a new Semisequent from an ordered collection of formulas.
+     * The provided collection must be redundancy free, i.e., the created sequent must be exactly
+     * the same as when creating the
      * sequent by subsequently inserting all formulas
      *
      * @param seqList list of sequent formulas
      */
-    public Semisequent(ImmutableList<SequentFormula> seqList) {
+    public Semisequent(Collection<SequentFormula> seqList) {
         assert !seqList.isEmpty();
-        this.seqList = seqList;
+        this.seqList = ImmutableList.fromList(seqList);
+    }
+
+    /**
+     * Create a new Semisequent from an ordered collection of formulas (possibly empty).
+     * The provided collection must be redundancy free, i.e., the created sequent must be exactly
+     * the same as when creating the
+     * sequent by subsequently inserting all formulas.
+     *
+     * @param seqList list of sequent formulas
+     */
+    public static Semisequent create(Collection<SequentFormula> seqList) {
+        if (seqList.isEmpty()) {
+            return EMPTY_SEMISEQUENT;
+        }
+        return new Semisequent(seqList);
     }
 
```

Comment:
Hi, just pushed another commit that restores the previous constructor (since `ImmutableList` doesn't implement `Collection`). That way it actually compiles successfully. And there is no performance hit.